backend/src/v1/linked_complaint_xref/linked_complaint_xref.service.ts

```diff
@@ -0,0 +1,110 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { CreateLinkedComplaintXrefDto } from "./dto/create-linked_complaint_xref.dto";
+import { LinkedComplaintXref } from "./entities/linked_complaint_xref.entity";
+import { InjectRepository } from "@nestjs/typeorm";
+import { DataSource, Repository } from "typeorm";
+import { HwcrComplaint } from "../hwcr_complaint/entities/hwcr_complaint.entity";
+import { Complaint } from "../complaint/entities/complaint.entity";
+
+@Injectable()
+export class LinkedComplaintXrefService {
+  @InjectRepository(LinkedComplaintXref)
+  private linkedComplaintXrefRepository: Repository<LinkedComplaintXref>;
+
+  private readonly logger = new Logger(LinkedComplaintXrefService.name);
+
+  constructor(private dataSource: DataSource) {}
+
+  async create(createLinkedComplaintXrefDto: CreateLinkedComplaintXrefDto): Promise<LinkedComplaintXref> {
+    const newLinkedComplaintXref = this.linkedComplaintXrefRepository.create(createLinkedComplaintXrefDto);
+    return newLinkedComplaintXref;
+  }
+
+  async findChildComplaints(parentComplaintId) {
+    try {
+      const builder = this.linkedComplaintXrefRepository
+        .createQueryBuilder("linkedComplaint")
+        .leftJoinAndMapOne(
+          "linkedComplaint.complaint",
+          Complaint,
+          "complaint",
+          "linkedComplaint.linked_complaint_identifier = complaint.complaint_identifier",
+        )
+        .leftJoin("complaint.complaint_status_code", "complaint_status")
+        .addSelect(["complaint_status.complaint_status_code", "complaint_status.short_description"])
+        .leftJoinAndMapOne(
+          "linkedComplaint.hwcr_complaint",
+          HwcrComplaint,
+          "hwcr_complaint",
+          "linkedComplaint.linked_complaint_identifier = hwcr_complaint.complaint_identifier",
+        )
+        .leftJoin("hwcr_complaint.species_code", "species_code")
+        .addSelect(["species_code.species_code", "species_code.short_description"])
+        .leftJoin("hwcr_complaint.hwcr_complaint_nature_code", "complaint_nature_code")
+        .addSelect(["complaint_nature_code.hwcr_complaint_nature_code", "complaint_nature_code.long_description"])
+        .where("linkedComplaint.complaint_identifier = :id", { id: parentComplaintId })
+        .andWhere("linkedComplaint.active_ind = :active", { active: true })
+        .orderBy("linkedComplaint.update_utc_timestamp", "DESC");
```

Comment:
I think this is going against an AC in CE-1136 that says that the complaints should be ordered from newest to older (incident_utc_datetime)